packages/netlify/src/ssr-function.ts

```diff
@@ -0,0 +1,54 @@
+import type { Context } from '@netlify/functions';
+import type { SSRManifest } from 'astro';
+import { App } from 'astro/app';
+import { applyPolyfills } from 'astro/app/node';
+
+applyPolyfills();
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface Args {}
+
+const clientAddressSymbol = Symbol.for('astro.clientAddress');
+
+export const createExports = (manifest: SSRManifest, _args: Args) => {
+	const app = new App(manifest);
+
+	function createHandler(integrationConfig: { cacheOnDemandPages: boolean }) {
+		return async function handler(request: Request, context: Context) {
+			const routeData = app.match(request);
+			Reflect.set(request, clientAddressSymbol, context.ip);
+
+			let locals: Record<string, unknown> = {};
+
+			if (request.headers.has('x-astro-locals')) {
+				locals = JSON.parse(request.headers.get('x-astro-locals')!);
```

Comment:
This operation can cause a parse error that we should catch and log the error that caused the issue.

Reply:
This header is set by the upstream edge function, so we can trust its format. If there's a parser error happening, there will be an exception showing in the Edge Function logs - that should be enough for debugging.